ir/type.def

```diff
@@ -62,6 +62,42 @@ inline bool operator>>(cstring s, IR::Direction &d) {
 }
 #end
 
+/// This represents a type that can unify with any other type.
+/// It is the original type of Dots (...) expressions.
+/// There is no syntax to represent this type.
+/// Treated like a type variable so that unification can assign it a value.
+class Type_Any : Type, ITypeVar {
```

Comment:
What is the difference between this and Type_DontCare?

Reply:
There is a literal for the don't care type, written as _. The unification rules are also different, don't care doesn't need to be unified, whereas ANY should always be resolved to a different, concrete type.

Reply:
Yes, the unification rules for Don't care are different. I thought about it.
But I think that having an extra class with a clear meaning is much simpler than overloading the meaning of the existing class and then reasoning about the different kinds of uses.
This way we are also sure we don't break anything that worked so far.